ada/testsuite/tests/name_resolution/primitive_return_type/test.adb

```diff
@@ -0,0 +1,18 @@
+procedure Main is
```

Comment:
Your code seems to suggest that this should also work in nested instantiations (eg. shed just the right amount of rebindings), however your test only contains one level of instantiation/nesting. Shouldn't you test with at least two ?